src/NuGet.Core/NuGet.Packaging/PackageArchiveReader.cs

```diff
@@ -384,7 +384,7 @@ public override Task<byte[]> GetArchiveHashAsync(HashAlgorithmName hashAlgorithm
         public override bool CanVerifySignedPackages(SignedPackageVerifierSettings verifierSettings)
         {
 #if IS_DESKTOP
-            return true;
+            return RuntimeEnvironmentHelper.IsWindows && !RuntimeEnvironmentHelper.IsMono;
```

Comment:
I think IsWindows is redundant. 

.NET Framework can only run under mono or on windows.